x-pack/legacy/plugins/actions/server/builtin_action_types/slack.test.ts

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { ActionType, Services } from '../types';
+import { ActionTypeRegistry } from '../action_type_registry';
+import { EncryptedSavedObjectsPlugin } from '../../../encrypted_saved_objects';
+import { SavedObjectsClientMock } from '../../../../../../src/core/server/mocks';
+import { validateActionTypeParams } from '../lib';
+import { validateActionTypeConfig } from '../lib';
+import { setIncomingWebhookImpl } from './slack';
+import { registerBuiltInActionTypes } from './index';
+import { MockIncomingWebhook } from './incoming_webhook.mock';
+import { taskManagerMock } from '../../../task_manager/task_manager.mock';
+
+const ACTION_TYPE_ID = 'kibana.slack';
+
+const NO_OP_FN = () => {};
+
+const services: Services = {
+  log: NO_OP_FN,
+  callCluster: async (path: string, opts: any) => {},
+  savedObjectsClient: SavedObjectsClientMock.create(),
+};
+
+function getServices(): Services {
+  return services;
+}
+
+let actionTypeRegistry: ActionTypeRegistry;
+let actionType: ActionType;
+
+const mockEncryptedSavedObjectsPlugin = {
```

Comment:
I created a new mock that we can now re-use `x-pack/legacy/plugins/encrypted_saved_objects/server/plugin.mock.ts`.